src/lm.jl

```diff
@@ -185,6 +185,31 @@ function adjr2(obj::LinearModel)
     1 - (1 - r²(obj))*(n-1)/(n-p)
 end
 
+struct FStatistic
+    nobs::Float64
+    ndims::Int64
+    f::Float64
+    fprob::Float64
+end
```

Comment:
```suggestion
struct FStatistic{T<:Real}
    df1::T
    df2::T
    f::T
    p::T
end
```

Note that the denominator degrees of freedom in a an F test do not have to be an integer (e.g. when using the Welch-Satterthwaite correction). Likewise, neither the numerator nor the denominator are exactly the number of observations nor the number of dimensions. They are degrees of freedom and there are various cases where it makes sense for these to be something than integers. (This also comes up in mixed models, where the effective degrees of freedom for test statistics is somewhat different than just the number of parameters.)

Reply:
Thank you, I did not know about non-integer DOF so mentioning Welch-Satterthwaite correction gave me something to study!

There's one problem in the resulting code, that the arguments to FStatistic don't have the same type. I solved this with 
`    FStatistic(promote(n, p, fstat, ccdf(fdist, abs(fstat)))...)` but I am not certain that's the right Julia way.